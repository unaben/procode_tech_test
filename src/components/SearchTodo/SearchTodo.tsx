import React, { FC } from "react";

type IFilteredTodoProps = {
  searchTerm: string;
  setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
};

const SearchTodo: FC<IFilteredTodoProps> = ({ searchTerm, setSearchTerm }) => {
  return (
    <form className="todo-input-item">
      <input
        value={searchTerm}
        onChange={(e) => setSearchTerm(e.target.value)}
        type="text"
        name="searchTerm"
        id="search-term"
        placeholder="Search todos...."
      />
    </form>
  );
};

export default SearchTodo;
