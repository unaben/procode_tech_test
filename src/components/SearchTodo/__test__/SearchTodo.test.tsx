import { render, screen } from "@testing-library/react";
import SearchTodo from "../SearchTodo";
import { Provider } from "react-redux";
import { store } from "../../../app/store";
import userEvent from "@testing-library/user-event";

describe("SearchTodo component", () => {
  it("should renders correctly", () => {
    const searchTerm = "";
    const mockFunction = jest.fn();
    const { container } = render(
      <Provider store={store}>
        <SearchTodo searchTerm={searchTerm} setSearchTerm={mockFunction} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("Should render input correctly", () => {
    const searchTerm = "";
    const mockFunction = jest.fn();
    render(
      <Provider store={store}>
        <SearchTodo searchTerm={searchTerm} setSearchTerm={mockFunction} />
      </Provider>
    );
    const inputEl = screen.getAllByRole("textbox");
    expect(inputEl).toHaveLength(1);
  });

  it("Should render a string when title Input is empty", () => {
    const searchTerm = "";
    const mockFunction = jest.fn();
    render(
      <Provider store={store}>
        <SearchTodo searchTerm={searchTerm} setSearchTerm={mockFunction} />
      </Provider>
    );
    const searchInput = screen.getByRole("textbox");
    userEvent.type(searchInput, "");

    expect(searchInput).toHaveValue("");
  });

  it("Should render a text when searchTerm Input is not empty", () => {
    const searchTerm = "Gym";
    const mockFunction = jest.fn();
    const user = userEvent.setup();
    render(
      <Provider store={store}>
        <SearchTodo searchTerm={searchTerm} setSearchTerm={mockFunction} />
      </Provider>
    );
    const searchInput = screen.getByRole("textbox");
    user.type(searchInput, "Gym");
    expect(searchInput).toHaveValue("Gym");
  });
});
