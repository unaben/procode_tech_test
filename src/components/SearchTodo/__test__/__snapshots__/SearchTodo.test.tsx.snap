// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTodo component should renders correctly 1`] = `
<div>
  <form
    class="todo-input-item"
  >
    <input
      id="search-term"
      name="searchTerm"
      placeholder="Search todos...."
      type="text"
      value=""
    />
  </form>
</div>
`;
